File: lua/tak.lua
local function tak(x, y, z)
    if y < x then
        return tak(
            tak(x - 1, y, z),
            tak(y - 1, z, x),
            tak(z - 1, x, y)
        )
    end
    return z
end

local args = {...}
local x = tonumber(args[1])
local y = tonumber(args[2])
local z = tonumber(args[3])
print(tak(x, y, z))

File: haskell/tak.hs
import System.Environment

tak :: Int -> Int -> Int -> Int
tak x y z
  | y < x = tak
    (tak (x - 1) y z)
    (tak (y - 1) z x)
    (tak (z - 1) x y)
  | otherwise = z

main :: IO ()
main = do
  args <- getArgs
  let [x, y, z] = map read args :: [Int]
  print $ tak x y z
